calyx-frontend/src/parser.rs

```diff
@@ -1029,71 +1040,111 @@ impl CalyxParser {
-                control,
-                attributes: attributes.add_span(span),
-                is_comb: false,
-            })
-        })
+            input.clone().into_children();
+            [
+                comb(_),
+                name_with_attribute((name, attributes)),
+                signature(sig),
+                cells(cells),
+                connections(connections)
+            ] => {
+                let (continuous_assignments, groups, static_groups) = connections;
+                let sig = sig.into_iter().map(|PortDef { name, width, direction, attributes }| {
+                    if let Width::Const { value } = width {
+                        Ok(PortDef {
+                            name,
+                            width: value,
+                            direction,
+                            attributes
+                        })
+                    } else {
+                        Err(input.error("Components cannot use parameters"))
+                    }
+                }).collect::<Result<_, _>>()?;
+                Ok(ComponentDef {
+                    name,
+                    signature: sig,
+                    cells,
+                    groups,
+                    static_groups,
+                    continuous_assignments,
+                    control: Control::empty(),
+                    attributes: attributes.add_span(span),
+                    is_comb: true,
+                    is_static: false,
+                    latency: None,
+                })
+            },
+            [
+                name_with_attribute((name, attributes)),
+                signature(sig),
+                cells(cells),
+                connections(connections),
+                control(control)
+            ] => {
+                let (continuous_assignments, groups, static_groups) = connections;
+                let sig = sig.into_iter().map(|PortDef { name, width, direction, attributes }| {
+                    if let Width::Const { value } = width {
+                        Ok(PortDef {
+                            name,
+                            width: value,
+                            direction,
+                            attributes
+                        })
+                    } else {
+                        Err(input.error("Components cannot use parameters"))
+                    }
+                }).collect::<Result<_, _>>()?;
+                Ok(ComponentDef {
+                    name,
+                    signature: sig,
+                    cells,
+                    groups,
+                    static_groups,
+                    continuous_assignments,
+                    control,
+                    attributes: attributes.add_span(span),
+                    is_comb: false,
+                    is_static: false,
+                    latency: None,
+                })
+            },
+            [
+                static_optional_latency(latency),
```

Comment:
What happens if someone specifies both `comb` and `static`? We should probably generate a reasonable here in the parser using `input.error` method and say that comb and static at the same time are not allowed. Also, worth moving the logic of parsing the keywords in front of a `component` into a separate method to deduplicate some of this stuff.